### PR TITLE
Update Adkernel adapter for 1.0 api

### PR DESCRIPTION
**Pull request by ckbo3hrk**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change
Migrating Adkernel adapter to api 1.0

- denis@adkernel.com
- [x] official adapter submission

### CODE REVIEW

File: modules/adkernelBidAdapter.js [hunk 1]
```diff
@@ -1,310 +1,191 @@
...
-     * Build ad-network specific endpoint url
-     */
-    function buildEndpointUrl(host) {
-      return `${window.location.protocol}//${host}/rtbg`;
+      });
+    });
+    return requests;
+  },
+  interpretResponse: function(serverResponse, request) {
+    /* istanbul ignore if */
```
Comment (dbemiller) [inline modules/adkernelBidAdapter.js]: Our linter won't fail on this... so you can remove it (along with the other istanbul suppressions you've got around here)

File: modules/adkernelBidAdapter.js [hunk 2]
```diff
@@ -1,310 +1,191 @@
...
+    let rtbRequest = JSON.parse(request.data.r);
+    let rtbImps = rtbRequest.imp;
+    let rtbBids = serverResponse.seatbid
+      .map(seatbid => seatbid.bid)
+      .reduce((a, b) => a.concat(b), []);
+
+    return rtbBids.map(rtbBid => {
+      let imp = rtbImps.find(imp => imp.id === rtbBid.impid);
+      let prBid = {
+        bidderCode: spec.code,
```
Comment (dbemiller) [inline modules/adkernelBidAdapter.js]: Our docs are bad... but you shouldn't send back the `bidderCode` here. The `bidderFactory` code does it automatically, and It breaks adapter aliasing if you do it yourself.

**Comment by dbemiller** [comment]
One more change, now that #1742 got merged.

The first argument to interpretResponse now looks like this:

{
  body: responseBody,
  headers: {
    get: function(header) { /* returns a header from the HTTP response */ }
  }
}
You'll have to update the spec so that it looks through that object as well now.

**Comment by ckbo3hrk** [comment]
@dbemiller everything is done

File: modules/adkernelBidAdapter.js [hunk 3]
```diff
@@ -1,310 +1,188 @@
...
-    // process individual bids
-    utils._each(bids, (bid) => {
-      if (!validateBidParams(bid.params)) {
-        utils.logError(`Incorrect configuration for adkernel bidder: ${bid.params}`);
-        bidmanager.addBidResponse(bid.placementCode, createEmptyBidObject(bid));
-      } else {
-        dispatcher.addImp(bid);
+    let result = [];
+    serverResponses.forEach(response => {
+      if (response.ext && response.ext.adk_usersync) {
```
Comment (dbemiller) [inline modules/adkernelBidAdapter.js]: In #1748, the server response format changed here. This was intentional... but I forgot to mention it.

Comment (ckbo3hrk) [reply modules/adkernelBidAdapter.js]: @dbemiller done

**Pull request merged**